contracts/RestrictedNORI.sol

```diff
@@ -375,7 +406,9 @@ contract RestrictedNORI is
       scheduleOwners: tokenHoldersLocal,
       quantitiesBurned: quantitiesToBurnForHolders
     });
-    _underlyingToken.transfer({to: toAccount, amount: quantityToRevoke});
+    if (!_underlyingToken.transfer({to: toAccount, amount: quantityToRevoke})) {
+      revert ERC20TransferFailed();
+    }
```

Comment:
Why aren't we using [SafeERC20](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/utils/SafeERC20.sol)

[This one](https://github.com/Vectorized/solady/blob/main/src/utils/SafeTransferLib.sol#L231) is another great alternative that's highly optimized

Reply:
First of all, fascinating medium article! I actually found myself wondering about whether there are ERC20 tokens out there that don't return anything at all while adding this check.... Turns out there are but they aren't *actually* ERC20s.

With regard to your actual comment here, I think you're confused about what we're doing here.  We are using a SafeERC20 in the sense that `_underlyingToken` here is literally `NORI` ... we originally decided not to bother checking the return value of `transfer` because we never intend to use `RestrictedNORI` with any underlying token other than `NORI`.   We may still want to take this approach, but given that *technically* the underlying token for `RestrictedNORI` IS configurable.... and by similar logic we recently decided to check the return values from transfers in the market contract, I just decided to 1) shut up Slither and 2) future-proof this a litttttle bit by doing the safest thing and just checking the return value.

We don't ever expect there to be a scenario where we're interacting with on of the BadTokens referenced in the article here, so that's way overkill to take into account.  But for the sake of best practice, I just threw the check in here.  Especially because I believe gas concerns are of lesser concern in this contract which will have far fewer interactions and is not buyer-facing.  Let me know if you'd rather I pull it out though.

Reply:
Makes sense given those constraints not to use the safe lib. Was thinking there might be a future where this also used a non nori token (super unlikely). I'd be in favor of removing the check here as it sets precedent for code style and the check itself is never needed.

Reply:
makes sense for me - gonna go ahead and remove the check.  if we're ever using a non-nori token as the underlying here I'm sure this entire contract will receive another round of scrutiny and we'll check return values then